### PR TITLE
Update to 1.10.1.post20200504175005

### PR DESCRIPTION
**Pull request by melund**
Update to newest version from pypi, and add small patch to support sphinx 4.

A very small change in sphinx means that the library fails with sphinx 4. I have [tried to contribute the fix upstream](https://foss.heptapod.net/doc-utils/cloud_sptheme/-/issues/45)  without any response, unfortunately. 

This PR fixes the problem so the packages can work with sphinx 4. 

**Edit:**
- Also fixed a number of linter issues in the recipe, updated to the newest recipe on pypi.

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Closes https://github.com/conda-forge/cloud_sptheme-feedstock/pull/9

### CODE REVIEW

**Comment by melund** [comment]
@conda-forge-admin, please rerender

**Comment by melund** [comment]
@jakirkham I have updated the library, fixed a number of small linter issues with the recipe. This is ready for review.

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,22 +7,25 @@ package:
...
+  number: 1
+  preserve_egg_dir: true
   script: {{ PYTHON }} -m pip install . -vv
 
+
 requirements:
   host:
     - pip
-    - python
+    - python >=3.6
```
Comment (jakirkham) [inline recipe/meta.yaml]: It doesn't appear that they have actually dropped Python 2. Could we please drop this constraint?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,22 +7,25 @@ package:
...
     - setuptools
 
   run:
-    - python
+    - python >=3.6
```
Comment (jakirkham) [inline recipe/meta.yaml]: Same question here

**Review by jakirkham — COMMENTED** [review]
Thanks Morten! 😄

Had one question below. Otherwise LGTM

**Review by jakirkham — APPROVED** [review]

**Comment by melund** [comment]
Thanks @jakirkham. I have changed the constraint to 2.7 instead.

**Pull request merged**